course/views.py

```diff
@@ -224,3 +229,49 @@ def form_valid(self, form):
             return self.render_to_response(self.get_context_data(
                 **group_info_context(enrollment.selected_group, self.profile)))
         return super().form_valid(form)
+
+
+class LanguageView(CourseInstanceMixin, BaseView):
+    
+    def post(self, request, *args, **kwargs):
+        LANGUAGE_PARAMETER = 'language'
+        
+        next = request.POST.get('next', request.GET.get('next'))
+        if ((next or not request.is_ajax()) and
+                not is_safe_url(url=next,
+                                allowed_hosts={request.get_host()}, 
+                                require_https=request.is_secure())):
+            next = request.META.get('HTTP_REFERER')
+            next = next and unquote(next)  # HTTP_REFERER may be encoded.
+            if not is_safe_url(url=next,
+                               allowed_hosts={request.get_host()},
+                               require_https=request.is_secure()):
+                next = '/'
+        response = HttpResponseRedirect(next) if next else HttpResponse(status=204)
+        if request.method == 'POST':
+            lang_code = request.POST.get(LANGUAGE_PARAMETER)
+            if lang_code and check_for_language(lang_code):
+                if next:
+                    next_trans = translate_url(next, lang_code)
+                    if next_trans != next:
+                        response = HttpResponseRedirect(next_trans)
+                    if request.user.is_authenticated:
+                        enrollment = self.instance.get_enrollment_for(self.request.user)
```

Comment:
Because this is a derived from `CourseInstanceMixin`, you can just use `self.user_in_course`